config/cmake/ConfigureChecks.cmake

```diff
@@ -745,7 +745,11 @@ if (HDF5_BUILD_FORTRAN)
 #else\n\
 #  define C_FLT128_DIG 0\n\
 #endif\n\
-#define C_LDBL_DIG DECIMAL_DIG\n\
+#ifdef DECIMAL_DIG\n\
```

Comment:
What compiler is this with? `DECIMAL_DIG` is C99, so I think we should be failing configure if C99 isn't supported rather than defining `C_LDBL_DIG` to 0 and hoping it will work.

Reply:
If you use -std=c99, it will be defined. HDF5 requires a minimum of C99. This option should be added for these older GNU compilers.